polkadot/runtime/rococo/src/lib.rs

```diff
@@ -1242,6 +1244,68 @@ impl pallet_asset_rate::Config for Runtime {
 	type BenchmarkHelper = runtime_common::impls::benchmarks::AssetRateArguments;
 }
 
+use frame_system::EnsureRootWithSuccess;
+use pallet_parameters::define_parameters;
+use sp_runtime::traits::ConstBool;
+
+/// Dynamic parameters that can be set by Root without a runtime upgrade.
+///
+/// All parameters expose metadata and docs.
+pub mod dynamic_params {
+	use super::*;
+
+	/// Dynamic params for [`pallet_nis`].
+	pub mod nis {
+		use super::*;
+
+		define_parameters! {
+			pub NisParams = {
+				/// Configures [`pallet_nis::Config::Target`].
+				#[codec(index = 0)]
+				NisTarget: Perquintill = Perquintill::zero(),
+			},
+			Pallet = pallet_parameters::Parameters::<Runtime>,
+			Aggregation = RuntimeParameters::Nis
+		}
+	}
+
+	/// Dynamic params for [`pallet_preimage`].
+	pub mod preimage {
+		use super::*;
+
+		define_parameters! {
+			pub PreimageParams = {
+				/// Configures the base deposit of noting a preimage.
+				#[codec(index = 0)]
+				BaseDeposit: Balance = deposit(2, 64),
+
+				/// Configures the per-byte deposit of noting a preimage.
+				#[codec(index = 1)]
+				ByteDeposit: Balance = deposit(0, 1),
+			},
+			Pallet = pallet_parameters::Parameters::<Runtime>,
+			Aggregation = RuntimeParameters::Preimage
+		}
+	}
+
+	define_aggregrated_parameters! {
```

Comment:
Seems like this should be achievable by `composite_enum`?

Reply:
Oh right, i should try this.